### PR TITLE
cache prototype - currently not working due to merge conflict

### PR DESCRIPTION
**Pull request by heejaechang**
this is just to show @pharring what I meant by simple time based MRU implicit cache.

### CODE REVIEW

File: src/EditorFeatures/Core/Implementation/Workspaces/ProjectCacheServiceFactory.ProjectCacheService.cs [hunk 1]
```diff
@@ -0,0 +1,296 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```
Comment (heejaechang) [inline src/EditorFeatures/Core/Implementation/Workspaces/ProjectCacheServiceFactory.ProjectCacheService.cs]: this file is only thing that matter.

File: src/EditorFeatures/Core/Implementation/Workspaces/ProjectCacheServiceFactory.ProjectCacheService.cs [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+                    cache.Count--;
+                    if (cache.Count == 0)
+                    {
+                        _activeCaches.Remove(key);
+                        cache.FreeOwnerEntries();
+                    }
+                }
+            }
+
+            private class ImplicitCacheMonitor : IdleProcessor
```
Comment (heejaechang) [inline src/EditorFeatures/Core/Implementation/Workspaces/ProjectCacheServiceFactory.ProjectCacheService.cs]: this should actually move into MRU cache

File: src/EditorFeatures/Core/Implementation/Workspaces/ProjectCacheServiceFactory.ProjectCacheService.cs [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+                        if (nodes[i].Data != null && nodes[i].LastTouchedInMS < expirationTimeInMS)
+                        {
+                            nodes[i] = default(Node);
+                        }
+                    }
+                }
+
+                public void Clear()
+                {
+                    for (var i = 0; i < nodes.Length; i++)
```
Comment (pharring) [inline src/EditorFeatures/Core/Implementation/Workspaces/ProjectCacheServiceFactory.ProjectCacheService.cs]:
```C#
Array.Clear(nodes, 0, nodes.Length);
```

**Comment by pharring** [comment]
This approach seems fine.

**Pull request closed without merging**